config.py

```diff
@@ -34,57 +34,76 @@
 c['slavePortnum'] = 9989
 
 from flocker_bb.password import generate_password
-from flocker_bb.ec2_buildslave import EC2BuildSlave
+from flocker_bb.ec2 import rackspace_slave, ec2_slave
 from buildbot.buildslave import BuildSlave
 
-cloudInit = FilePath(__file__).sibling("slave").child("cloud-init.sh").getContent()
+cloudInit = FilePath(__file__).sibling("slave").child(
+    "cloud-init.sh").getContent()
 
 c['slaves'] = []
 SLAVENAMES = {}
 
+
+def get_cloud_init(name, base, password, provider, privateData, slavePortnum):
+    return cloudInit % {
+        "github_token": privateData['github']['token'],
+        "coveralls_token": privateData['coveralls']['token'],
+        "name": name,
+        "base": base,
+        "password": password,
+        "FLOCKER_FUNCTIONAL_TEST_CLOUD_PROVIDER": provider,
+        'buildmaster_host': privateData['buildmaster']['host'],
+        'buildmaster_port': slavePortnum,
+        'acceptance.yml': privateData['acceptance'].get('config', ''),
+        'acceptance-ssh-key': privateData['acceptance'].get('ssh-key', ''),
+    }
```

Comment:
This is mostly the same as what was being passed inline to the EC2 slave previously - but with the new `FLOCKER_FUNCTIONAL_TEST_CLOUD_PROVIDER` variable interpolated so the cloud-init script actually knows what cloud we're on.